imap/annotate.c

```diff
@@ -1229,7 +1215,7 @@ static int annotatemore_findall_full(const char *pattern, /* internal */
         struct findall_data data = { .mbentry = mbentry, .is_exactmatch = 1 };
         r = _findall(&data, &frock);
     }
-    else if (!*pattern) {
+    else if (!pattern) {
```

Comment:
I'm not sure if we should be looking for BOTH here - NULL or blank string.

Reply:
Empty string in the previous code was used to signal server entries, but that signaling has changed.  Having said that I have to issue with checking for both.